ClosedXML/Excel/Caching/XLRepositoryBase.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Collections;
+
+namespace ClosedXML.Excel.Caching
+{
+    internal abstract class XLRepositoryBase : IXLRepository
+    {
+        public abstract void Clear();
+    }
+
+    internal abstract class XLRepositoryBase<Tkey, Tvalue> : XLRepositoryBase, IXLRepository<Tkey, Tvalue>
+        where Tkey : struct, IEquatable<Tkey>
+        where Tvalue : class
+    {
+        const int CONCURRENCY_LEVEL = 4;
+        const int INITIAL_CAPACITY = 1000;
+
+        private readonly ConcurrentDictionary<Tkey, WeakReference> _storage;
```

Comment:
Not yet, although I thought this can be used for propagating style changes from parent to children in parallel.
Potentially, this may allow working with multiple workbooks in independent threads (as long as they do not interact it is more or less safe, but if they share global style repositories with ordinary dictionaries the result could be unpredictable). Then, with an ordinary dictionary unit tests running in parallel would randomly fail. 
And finally, ConcurrentDictionary may be safely modified when it is enumerated, and this helped me with #766.